pkgs/by-name/gi/github-desktop/package.nix

```diff
@@ -1,107 +1,200 @@
+    yarnLock = finalAttrs.src + "/app/yarn.lock";
+    hash = "sha256-FTKqpay0ysConZ0LoAAgSxiNJqO1te7G20XHRzQj0WE=";
+  };
+
+  cacheDesktopNotifications = fetchYarnDeps {
+    yarnLock = finalAttrs.src + "/vendor/desktop-notifications/yarn.lock";
+    hash = "sha256-2Hd8baxxoseWVTSChP6AKIODOlOAHK0ALAGQr2gpoQM=";
+  };
+
+  dontYarnInstallDeps = true;
+
+  postConfigure = ''
+    yarnOfflineCache="$cacheRoot" yarnConfigHook
+
+    pushd app
+    yarnOfflineCache="$cacheApp" yarnConfigHook
+    popd
+
+    # This is insanely hacky, I don't even really understand why this has to be
+    # the way it is...
+    pushd app/node_modules/desktop-notifications
+    yarnOfflineCache="$cacheDesktopNotifications" yarnConfigHook
+    npm_config_nodedir="${nodejs}" yarn --offline install
+    popd
+
+    patchShebangs node_modules
+    patchShebangs app/node_modules
+    patchShebangs app/node_modules/desktop-notifications/node_modules
+
+    # use git from nixpkgs instead of an automatically downloaded one by dugite
+    echo > app/node_modules/dugite/script/download-git.js
+    cp -r ${git} app/node_modules/dugite/git
+    chmod -R +w app/node_modules/dugite/git
+
+    # yarnConfigHook never runs lifecycle scripts so yeah this is what you get
+    # I guess
+    # This _has_ to be simpler no?
+    for native in fs-admin keytar desktop-trampoline; do
+      npm_config_nodedir="${nodejs}" yarn --offline --cwd app/node_modules/$native build
+    done
+
+    yarn compile:script
+
+    # Because ts-loader isn't set and this package doesn't get compiled
+    # automatically
+    pushd app/node_modules/windows-argv-parser
+    tsc
+    popd
+
+    cp -r ${electron.dist} electron-dist
+    chmod -R u+w electron-dist
+
+    pushd electron-dist
+    zip -0Xqr ../electron.zip .
+    popd
+
+    rm -r electron-dist
+
+    substituteInPlace node_modules/electron-packager/src/index.js \
+      --replace-fail "await this.getElectronZipPath(downloadOpts)" "\"$(pwd)/electron.zip\""
   '';
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "github-desktop";
+      desktopName = "GitHub Desktop";
+      comment = "Focus on what matters instead of fighting with Git";
+      exec = "github-desktop %U";
+      icon = "github-desktop";
+      terminal = false;
+    })
+  ];
+
   installPhase = ''
     runHook preInstall
-    cp -R $TMP/github-desktop/usr/share $out/
-    cp -R $TMP/github-desktop/usr/lib/github-desktop/* $out/opt/
-    ln -sf $out/opt/github-desktop $out/bin/github-desktop
-    runHook postInstall
-  '';
 
-  preFixup = ''
-    gappsWrapperArgs+=(
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-wayland-ime=true}}"
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libglvnd ]}
-    )
+    mkdir -p $out/share/github-desktop
+
+    # transpose [name][size] into [size][name]
+    for icon in app/static/logos/*.png; do
+      basename="$(basename "$icon")"
+      size="''${basename%.png}"
```

Comment:
You can just do `size="$(basename "$icon" .png)"` I think. A nice feature of the `basename` command that is good to know.